sdks/python/apache_beam/io/textio.py

```diff
@@ -390,7 +390,7 @@ def __init__(self,
   def open(self, temp_path):
     file_handle = super(_TextSink, self).open(temp_path)
     if self._header is not None:
-      file_handle.write(self._header)
+      file_handle.write(self.coder.encode(self._header))
```

Comment:
This will work well when `coder` is initialized to default value `coders.ToStringCoder()`, however if the user passes a different coder to `WriteToText` PTransform, which instantiates `_TextSink`, such coder may not encode text correctly.
The fastest way to fix this would be to use:
`file_handle.write(coders.ToStringCoder().encode(self._header))`.
Alternatively we could introduce a helper function `as_bytes`  for these purposes, and put it somewhere a handy place.

Reply:
@tvalentyn Thank you for your review.
It's right that incorrect coder causes errors.
However, `coder` is also used to encode lines at `apache_beam.io.filebasedsink.FileBasedSink.write_record`.
FileBasedSink is parent class of `textio._TextSink`.
Encoding header without the `coder` will make header and lines to be encoded with different coders or function, is that OK?

Reply:
This should be fine, because while records may have different underlying types, header should always be a string according to the Api definition.